tests/test_pasqal.py

```diff
@@ -190,15 +196,82 @@ def test_remote_results(fixt, mock_batch, with_job_id):
 
     assert hasattr(remote_results, "_results")
 
+    fixt.mock_cloud_sdk.get_batch.reset_mock()
+    available_results = remote_results.get_available_results("id")
+    assert available_results == {
+        job.id: SampledResult(
+            atom_order=("q0", "q1", "q2", "q3"),
+            meas_basis="ground-rydberg",
+            bitstring_counts=job.result,
+        )
+        for job in select_jobs
+    }
+
+
+def test_partial_results():
+    batch = MockBatch(
+        status="RUNNING",
+        ordered_jobs=[
+            _MockJob(),
+            _MockJob(status="RUNNING", result=None),
+        ],
+    )
+
+    fixt = mock_pasqal_cloud_sdk(batch)
+
+    remote_results = RemoteResults(
+        batch.id,
+        fixt.pasqal_cloud,
+    )
+
+    fixt.mock_cloud_sdk.get_batch.reset_mock()
+    with pytest.raises(RemoteResultsError):
+        remote_results.results
+    available_results = remote_results.get_available_results(batch.id)
+    assert available_results == {
+        job.id: SampledResult(
+            atom_order=("q0", "q1", "q2", "q3"),
+            meas_basis="ground-rydberg",
+            bitstring_counts=job.result,
+        )
+        for job in batch.ordered_jobs
+        if job.result is not None
+    }
+
+    batch = MockBatch(
+        status="DONE",
+        ordered_jobs=[
+            _MockJob(),
+            _MockJob(status="DONE", result=None),
+        ],
+    )
+
+    fixt = mock_pasqal_cloud_sdk(batch)
+    remote_results = RemoteResults(
+        batch.id,
+        fixt.pasqal_cloud,
+    )
+
+    with pytest.raises(RemoteResultsError):
+        remote_results.results
+    available_results = remote_results.get_available_results(batch.id)
+    assert available_results == {
+        job.id: SampledResult(
+            atom_order=("q0", "q1", "q2", "q3"),
+            meas_basis="ground-rydberg",
+            bitstring_counts=job.result,
+        )
+        for job in batch.ordered_jobs
+        if job.result is not None
+    }
+
```

Comment:
If you could, some checks that the expected methods in the mock are being called would be nice

Reply:
Done